src/w3d/renderer/render2dsentence.cpp

```diff
@@ -207,8 +306,149 @@ void FontCharsClass::Update_Current_Buffer(int char_width)
     }
 }
 
+bool FontCharsClass::Locate_Font(const char *font_name, StringClass &font_file_path)
+{
+#if defined BUILD_WITH_FONTCONFIG
+    if (m_fc == nullptr) {
+        return false;
+    }
+
+    FcPattern *pattern = FcNameParse(reinterpret_cast<const FcChar8 *>(font_name));
+    FcConfigSubstitute(m_fc, pattern, FcMatchPattern);
+    FcDefaultSubstitute(pattern);
+
+    // Find the font
+    FcResult result;
+    FcPattern *font = FcFontMatch(m_fc, pattern, &result);
+    if (font != nullptr) {
+        FcChar8 *file = NULL;
+        if (FcPatternGetString(font, FC_FILE, 0, &file) == FcResultMatch) {
+            // Save the file to another std::string
+            font_file_path = (char *)file;
+        }
+        FcPatternDestroy(font);
+    }
+
+    FcPatternDestroy(pattern);
+    return result == FcResultMatch;
+#elif defined PLATFORM_WINDOWS
+    const char *font_registry_key = "Software\\Microsoft\\Windows NT\\CurrentVersion\\Fonts";
+    HKEY hkey;
+    LONG result;
+
+    // Open Windows font registry key
+    result = RegOpenKeyEx(HKEY_LOCAL_MACHINE, font_registry_key, 0, KEY_READ, &hkey);
+    if (result != ERROR_SUCCESS) {
+        return false;
+    }
+
+    DWORD max_value_name_size, max_value_data_size;
+    result = RegQueryInfoKey(hkey, 0, 0, 0, 0, 0, 0, 0, &max_value_name_size, &max_value_data_size, 0, 0);
+    if (result != ERROR_SUCCESS) {
+        return false;
+    }
+
+    DWORD value_index = 0;
+    char *value_name = new char[max_value_name_size];
+    LPBYTE value_data = new BYTE[max_value_data_size];
+    DWORD value_name_size, value_data_size, value_type;
+    StringClass font_file;
+
+    // Look for a matching font name
+    do {
+        value_data_size = max_value_data_size;
+        value_name_size = max_value_name_size;
+
+        result = RegEnumValue(hkey, value_index, value_name, &value_name_size, 0, &value_type, value_data, &value_data_size);
+
+        value_index++;
+
+        if (result != ERROR_SUCCESS || value_type != REG_SZ) {
+            continue;
+        }
+
+        // Found a match
+        DWORD max_count = value_name_size < strlen(font_name) ? value_name_size : strlen(font_name);
+        if (strnicmp(font_name, value_name, max_count) == 0) {
+            font_file = reinterpret_cast<char *>(value_data);
+            break;
+        }
+    } while (result != ERROR_NO_MORE_ITEMS);
+
+    delete[] value_name;
+    delete[] value_data;
+
+    RegCloseKey(hkey);
+
+    if (font_file.Is_Empty()) {
+        return false;
+    }
+
+    // Build full font file path
+    char windows_dir[MAX_PATH];
+    if (GetWindowsDirectory(windows_dir, MAX_PATH) == 0) {
```

Comment:
Perhaps do this test much earlier? No need to do all this stuff above when this already fails.